### PR TITLE
Boats: Use driver name string instead of objectref

### PR DESCRIPTION
**Pull request by paramat**
Requested by @SmallJoker in https://github.com/minetest/minetest_game/pull/2140#issuecomment-401783486

### CODE REVIEW

**Comment by tenplus1** [comment]
How about boats being destroyed when hitting land too hard and dropping wood: https://notabug.org/TenPlus1/boats/src/master/init.lua#L255

**Comment by SmallJoker** [comment]
@tenplus1 That's a feature which belongs into another PR (also so that it can be discussed properly).

**Comment by tenplus1** [comment]
@SmallJoker - no probs, was asking to see if it was interesting enough to add a pull :P

**Comment by paramat** [comment]
I would be -1 for that feature anyway, watched many MC videos where players get fed up with that.

**Comment by paramat** [comment]
Testing on a local server with a client driver leaving the game while driving, a few of these popup:
```
2018-07-14 23:07:54: WARNING[Server]: ScriptApiBase::objectrefGetOrCreate(): Pushing ObjectRef to removed/deactivated object, this is probably a bug.
```
Will attempt to remove those using a player 'on leave' callback.

**Comment by paramat** [comment]
Yes that works, avoids those warnings and means i don't need my alternative solution.

**Comment by paramat** [comment]
Updated and tested.

File: mods/boats/init.lua [hunk 1]
```diff
@@ -132,35 +141,35 @@ end
...
 	if self.driver then
-		local driver_name = self.driver:get_player_name()
-		local ctrl = self.driver:get_player_control()
-		if ctrl.up and ctrl.down then
-			if not self.auto then
-				self.auto = true
-				minetest.chat_send_player(driver_name,
-					"[boats] Cruise on")
+		local driver_objref = minetest.get_player_by_name(self.driver)
+		if driver_objref then
```
Comment (SmallJoker) [inline mods/boats/init.lua]: How about the `else` case? Didn't the previous code (similar to what is in `boat.on_detach_child` now) help to reset the driver name when it became invalid?

Comment (paramat) [reply mods/boats/init.lua]: Using the else case here is what caused https://github.com/minetest/minetest_game/pull/2175#issuecomment-405053172

**Comment by paramat** [comment]
Approved by SmallJoker http://irc.minetest.net/minetest-dev/2018-07-15#i_5359558

**Pull request merged**